pkg/operator/vspherecontroller/vspherecontroller.go

```diff
@@ -633,13 +637,13 @@ func (c *VSphereController) createCSIConfigMap(
 
 	datastoreURL := defaultDatastore.Summary.Url
 
-	requiredCM, err := c.applyClusterCSIDriverChange(infra, cfg, clusterCSIDriver, datastoreURL)
+	requiredSecret, err := c.applyClusterCSIDriverChange(infra, cfg, clusterCSIDriver, datastoreURL)
 	if err != nil {
 		return err
 	}
 
 	// TODO: check if configMap has been deployed and set appropriate conditions
-	_, _, err = resourceapply.ApplyConfigMap(ctx, c.kubeClient.CoreV1(), syncContext.Recorder(), requiredCM)
+	_, _, err = resourceapply.ApplySecret(ctx, c.kubeClient.CoreV1(), syncContext.Recorder(), requiredSecret)
```

Comment:
What happens in upgraded clusters? Do we need to remove the previously configured configmap?

Reply:
Added logic to remove legacy ConfigMap